### ISSUE TITLE
Investigate use of `arguments` in logger.

### ISSUE THREAD

**Issue by ofrobots** [GoogleCloudPlatform/cloud-diagnostics-common-nodejs#6]

**Comment by cristiancavalli** [comment]
@matthewloring @ofrobots can this be closed? Seems like it was implemented already

**Comment by matthewloring** [comment]
This bug was opened to investigate the performance impact of arguments used here: https://github.com/GoogleCloudPlatform/cloud-diagnostics-common-nodejs/blob/master/lib/logger.js#L99. If it is causing deopts or poor performance we may want to rework this section.

**Comment by matthewloring** [comment]
We observed deopts in `join` (https://github.com/GoogleCloudPlatform/cloud-diagnostics-common-nodejs/blob/master/lib/logger.js#L73) and `slice` (https://github.com/GoogleCloudPlatform/cloud-diagnostics-common-nodejs/blob/master/lib/logger.js#L99) but nothing associated with `arguments`. It will be difficult to avoid these deopts as logs can be called on arbitrary numbers of arguments (though it would be good to investigate V8 changes to avoid these). We can do a more detailed analysis of logging performance if needed. Closing this for now.
